File: Blind75/5. Maximum Sum in Contiguous Array/Maximum_Sum_in_Contiguous_Array.py
def maxSumSubArray(numbers: list[int]) -> int:
    result = float('-inf')
    sum = 0
    
    for number in numbers:
        sum += number
        result = max(sum, result)
        if sum < 0:
            sum = 0 
            
    return result

test_cases = [[-1,5,-3,9,-11], [9], [1,2,3,4]]
for nums in test_cases:
    print(maxSumSubArray(nums))
    

File: Blind75/Maximum Sum in Contiguous Array/Maximum_Sum_in_Contiguous_Array.js
function maxSumSubArray(numbers){
    let result = -Infinity
    let sum = 0

    for (let num of numbers){
        sum += num
        result = Math.max(sum, result)

        if(sum < 0){
            sum = 0
        }
    }
    return result
}

const testCases =  [[-1,5,-3,9,-11], [9], [1,2,3,4]]
for(const nums of testCases){
    console.log(maxSumSubArray(nums))
}